Composer/packages/client/src/pages/publish/PublishDialog.tsx

```diff
@@ -0,0 +1,129 @@
+      className: 'botName',
+      fieldName: 'name',
+      minWidth: 70,
+      maxWidth: 90,
+      data: 'string',
+      onRender: (item: IBotStatus) => {
+        return <div css={{ alignItems: 'center', display: 'flex', height: '32px' }}>{item.name}</div>;
+      },
+      isPadded: true,
+    },
+    {
+      key: 'publishTarget',
+      name: formatMessage('Publish target'),
+      className: 'publishtarget',
+      fieldName: 'publishTarget',
+      minWidth: 70,
+      maxWidth: 90,
+      data: 'string',
+      onRender: (item: IBotStatus) => {
+        return (
+          <div
+            css={{
+              backgroundColor: '#DDF3DB',
+              alignItems: 'center',
+              display: 'flex',
+              height: '32px',
+              paddingLeft: '8px',
+            }}
+          >
+            {item.publishTarget}
+          </div>
+        );
+      },
+      isPadded: true,
+    },
+    {
+      key: 'comment',
+      name: formatMessage('Comments'),
+      className: 'comment',
+      fieldName: 'comment',
+      minWidth: 70,
+      maxWidth: 90,
+      data: 'string',
+      onRender: (item: IBotStatus) => {
+        // message for each publish bot
+        return (
+          <TextField
+            placeholder={formatMessage('Write your message')}
+            value={item.comment}
+            onChange={(e, newValue) => {
+              const newItems = showItems.map((obj) => {
+                if (obj.id === item.id) {
+                  obj.comment = newValue;
+                }
+                return obj;
+              });
+              setShowItems(newItems);
+            }}
+          />
+        );
+      },
+      isPadded: true,
+    },
+  ];
+  const publishDialogProps = {
+    title: formatMessage('Publish'),
+    type: DialogType.normal,
+    subText: formatMessage('You are about to publish your bot to the profile below. Do you want to proceed?'),
+  };
+  const cleanComments = () => {
+    const cleanedItems = showItems.map((item) => {
+      item.comment = '';
+      return item;
+    });
+    setShowItems(cleanedItems);
+  };
+  const submit = async () => {
+    props.onDismiss();
+    await props.onSubmit(showItems);
+    cleanComments();
+  };
+  return showItems && showItems.length > 0 ? (
+    <Dialog
+      dialogContentProps={publishDialogProps}
+      hidden={false}
+      modalProps={{ isBlocking: true, styles: { main: { maxWidth: '1063px !important' } } }}
+      onDismiss={() => {
+        cleanComments();
+        props.onDismiss();
+      }}
+    >
+      <Fragment>
+        <DetailsList checkboxVisibility={CheckboxVisibility.hidden} columns={columns} items={showItems} />
+
+        <DialogFooter>
+          <DefaultButton
+            text={formatMessage('Cancel')}
+            onClick={() => {
+              cleanComments();
+              props.onDismiss();
+            }}
+          />
+          <PrimaryButton text={formatMessage('Okay')} onClick={submit} />
```

Comment:
is this copy correct Okay? seems wrong

Reply:
![image](https://user-images.githubusercontent.com/5860999/100058782-e6e05180-2e64-11eb-8d37-ad603c83148a.png)
It is correct